File: ncDados/dEfd.vb
Namespace nsEFD

  Public Class ColC190
    Inherits List(Of dC190)
  End Class

  Public Class dC190

    Private _reg As String
    Private _cst_icms As Nullable(Of Integer)
    Private _cfop As Nullable(Of Integer)
    Private _aliq_icms As Nullable(Of Decimal)
    Private _vl_opr As Nullable(Of Decimal)
    Private _vl_bc_icms As Nullable(Of Decimal)
    Private _vl_icms As Nullable(Of Decimal)
    Private _vl_bc_icms_st As Nullable(Of Decimal)
    Private _vl_icms_st As Nullable(Of Decimal)
    Private _vl_red_bc As Nullable(Of Decimal)
    Private _vl_ipi As Nullable(Of Decimal)
    Private _cod_obs As String

    Public Property reg() As String
      Get
        Return _reg
      End Get
      Set(ByVal value As String)
        _reg = value
      End Set
    End Property

    Public Property cst_icms() As Nullable(Of Integer)
      Get
        Return _cst_icms
      End Get
      Set(ByVal value As Nullable(Of Integer))
        _cst_icms = value
      End Set
    End Property

    Public Property cfop() As Nullable(Of Integer)
      Get
        Return _cfop
      End Get
      Set(ByVal value As Nullable(Of Integer))
        _cfop = value
      End Set
    End Property

    Public Property aliq_icms() As Nullable(Of Decimal)
      Get
        Return _aliq_icms
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _aliq_icms = value
      End Set
    End Property

    Public Property vl_opr() As Nullable(Of Decimal)
      Get
        Return _vl_opr
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _vl_opr = value
      End Set
    End Property

    Public Property vl_bc_icms() As Nullable(Of Decimal)
      Get
        Return _vl_bc_icms
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _vl_bc_icms = value
      End Set
    End Property

    Public Property vl_icms() As Nullable(Of Decimal)
      Get
        Return _vl_icms
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _vl_icms = value
      End Set
    End Property

    Public Property vl_bc_icms_st() As Nullable(Of Decimal)
      Get
        Return _vl_bc_icms_st
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _vl_bc_icms_st = value
      End Set
    End Property

    Public Property vl_icms_st() As Nullable(Of Decimal)
      Get
        Return _vl_icms_st
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _vl_icms_st = value
      End Set
    End Property

    Public Property vl_red_bc() As Nullable(Of Decimal)
      Get
        Return _vl_red_bc
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _vl_red_bc = value
      End Set
    End Property

    Public Property vl_ipi() As Nullable(Of Decimal)
      Get
        Return _vl_ipi
      End Get
      Set(ByVal value As Nullable(Of Decimal))
        _vl_ipi = value
      End Set
    End Property

    Public Property cod_obs() As String
      Get
        Return _cod_obs
      End Get
      Set(ByVal value As String)
        _cod_obs = value
      End Set
    End Property

  End Class

  Public Class dEfd

  End Class

End Namespace

File: Modelos/dEfd.cs
using System;
using System.Collections.Generic;

namespace Modelos
{
    public class ColC190 : List<dC190>
    {
    }

    public class dC190
    {
        private string _reg;
        private int? _cst_icms;
        private int? _cfop;
        private decimal? _aliq_icms;
        private decimal? _vl_opr;
        private decimal? _vl_bc_icms;
        private decimal? _vl_icms;
        private decimal? _vl_bc_icms_st;
        private decimal? _vl_icms_st;
        private decimal? _vl_red_bc;
        private decimal? _vl_ipi;
        private string _cod_obs;

        public string reg
        {
            get => _reg;
            set => _reg = value;
        }

        public int? cst_icms
        {
            get => _cst_icms;
            set => _cst_icms = value;
        }

        public int? cfop
        {
            get => _cfop;
            set => _cfop = value;
        }

        public decimal? aliq_icms
        {
            get => _aliq_icms;
            set => _aliq_icms = value;
        }

        public decimal? vl_opr
        {
            get => _vl_opr;
            set => _vl_opr = value;
        }

        public decimal? vl_bc_icms
        {
            get => _vl_bc_icms;
            set => _vl_bc_icms = value;
        }

        public decimal? vl_icms
        {
            get => _vl_icms;
            set => _vl_icms = value;
        }

        public decimal? vl_bc_icms_st
        {
            get => _vl_bc_icms_st;
            set => _vl_bc_icms_st = value;
        }

        public decimal? vl_icms_st
        {
            get => _vl_icms_st;
            set => _vl_icms_st = value;
        }

        public decimal? vl_red_bc
        {
            get => _vl_red_bc;
            set => _vl_red_bc = value;
        }

        public decimal? vl_ipi
        {
            get => _vl_ipi;
            set => _vl_ipi = value;
        }

        public string cod_obs
        {
            get => _cod_obs;
            set => _cod_obs = value;
        }
    }

    public class dEfd
    {
    }

}
